spartan/devnet-smoke-tests/process_token_contracts.sh

```diff
@@ -0,0 +1,84 @@
+      private_balance=$(aztec-wallet simulate balance_of_private -ca "$token_address" --args "$current_user_address" -f "$current_user_address" --node-url $NODE_URL | grep "Simulation result:" | awk '{print $3}')
+
+      echo "Account $current_user_address for token address $token_address public balance is ${public_balance}"
+      echo "Account $current_user_address for token address $token_address private balance is ${private_balance}"
+
+      setup_proven=true
+    else
+      echo "Skipping minting for $current_user_address from token $token_address - already initialized"
+    fi
+  done
+
+  other_accounts=$(jq -c --arg curr "$current_user_address" '.accounts[] | select(.address != $curr)' state.json)
+
+  other_accounts_count=$(echo "$other_accounts" | jq -s 'length')
+  if [ "$other_accounts_count" -eq 0 ]; then
+      echo "No other accounts found, exiting..."
+      break
+  fi
+
+  # We then process each contract
+  jq -c '.contracts | map(select(.type=="token"))[]' state.json | while read -r contract; do
+    token_address=$(echo "$contract" | jq -r '.address')
+
+    public_balance=$(aztec-wallet simulate balance_of_public -ca "$token_address" --args $current_user_address -f $current_user_address --node-url $NODE_URL | grep "Simulation result:" | awk '{print $3}' | tr -d 'n')
+    private_balance=$(aztec-wallet simulate balance_of_private -ca "$token_address" --args $current_user_address -f $current_user_address --node-url $NODE_URL | grep "Simulation result:" | awk '{print $3}' | tr -d 'n')
+
+    echo "Account $current_user_address for token address $token_address public balance is ${public_balance}"
+    echo "Account $current_user_address for token address $token_address private balance is ${private_balance}"
+
+    # For every other account, we distribute 1/2 of our private and public balances to them
+    amount_to_transfer=$(($public_balance/2/$other_accounts_count))
+    echo "Amount to transfer $amount_to_transfer"
+
+    echo "$other_accounts" | while read -r other_account; do
+      other_address=$(echo "$other_account" | jq -r '.address')
+
+      echo "Token address: $token_address, Current address: $current_user_address, Other address: $other_address"
+
+      aztec-wallet $([ "$process_proven" = "false" ] && echo "-p native" || echo "") send transfer_in_public -ca "$token_address" --args "$current_user_address" "$other_address" $amount_to_transfer 0 -f "$current_user_address" --node-url $NODE_URL $([ "$use_sponsored_fpc" = "true" ] && echo "$SPONSORED_FPC_PAYMENT_METHOD" || echo "")
+      aztec-wallet $([ "$process_proven" = "false" ] && echo "-p native" || echo "") send transfer_in_private -ca "$token_address" --args "$current_user_address" "$other_address" $amount_to_transfer 0 -f "$current_user_address" --node-url $NODE_URL $([ "$use_sponsored_fpc" = "true" ] && echo "$SPONSORED_FPC_PAYMENT_METHOD" || echo "")
```

Comment:
'process' seems fairly generic to find hidden here that what we're doing in this test is transferring some tokens. It'd be nice to have some description of what each test actually tests.